consensus/spos/bls/subroundSignature.go

```diff
@@ -386,10 +403,11 @@ func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
 
 			numMultiKeysSignaturesSent++
 		}
+		// with the equivalent messages feature on, the leader signature is sent on subroundBlock, thus we should update its status here as well
 		sr.sentSignatureTracker.SignatureSent(pkBytes)
 
-		isLeader := idx == spos.IndexOfLeaderInConsensusGroup
-		ok := sr.completeSignatureSubRound(pk, isLeader)
+		shouldWaitForAllSigsAsync := isKeyLeader && !isFlagActive
+		ok := sr.completeSignatureSubRound(pk, shouldWaitForAllSigsAsync)
```

Comment:
if `isFlagActive` if false shouldn't we wait for all sigs also if `isMultiKeyLeader` is true?

Reply:
when the flag is not active(as before), we only want to start one waiting go routine, thus we start it only for the specific managed key that is waiting leader

after flag activation, we don't want to wait here.. this will happen on subroundEndRound